sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/PagedResponse.java

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.http.models;
+
+import io.clientcore.core.util.binarydata.BinaryData;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Response of a REST API that returns page.
+ *
+ * @see Response
+ *
+ * @param <T> The type of items in the page.
+ */
+public final class PagedResponse<T> implements Response<List<T>> {
+
+    private final HttpRequest request;
+    private final int statusCode;
+    private final HttpHeaders headers;
+    private final List<T> items;
+    private final String nextLink;
+    private final BinaryData body;
+
+    /**
+     * Creates a new instance of the PagedResponse type.
+     *
+     * @param request The HttpRequest that was sent to the service whose response resulted in this response.
+     * @param statusCode The status code from the response.
+     * @param headers The headers from the response.
+     * @param body The body from the response.
+     * @param items The items returned from the service within the response.
+     * @param nextLink The next page reference returned from the service, to enable future requests to pick up
+     *      from the same place in the paged iteration.
+     */
+    public PagedResponse(HttpRequest request, int statusCode, HttpHeaders headers, BinaryData body, List<T> items,
```

Comment:
Since we have the raw BinaryData body, do we need `List<T> items` too? Do we need to eagerly deserialize the list?

Reply:
At present, removing the `List<T>` in ctor does not provide benefits.

The de-serialization already happens to the response, e.g. here the `TodoPage`
https://github.com/Azure/azure-sdk-for-java/blob/d7e8c4e1e54ef1654dffb138403ce54ebb2894b1/sdk/clientcore/core/src/test/java/io/clientcore/core/http/models/PagedIterableTests.java#L56-L58